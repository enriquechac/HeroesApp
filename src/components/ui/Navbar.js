import React, { useContext } from 'react';
import { Link, NavLink, useHistory } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const Navbar = () => {
    const { user, dispatch } = useContext(AuthContext);

    const history = useHistory();
    // console.log(history);

    const handleLogout = () => {
        // console.log('Click!');
        history.replace('/login');
        dispatch({
            type: types.logout,
        });
    };

    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <Link className="navbar-brand" to="/">
                Asociaciones
            </Link>

            <div className="navbar-collapse collapse">
                <div className="navbar-nav">
                    <NavLink
                        activeClassName="active"
                        className="nav-item nav-link"
                        exact
                        to="/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink
                        activeClassName="active"
                        className="nav-item nav-link"
                        exact
                        to="/dc"
                    >
                        DC
                    </NavLink>
                    <NavLink
                        activeClassName="active"
                        className="nav-item nav-link"
                        exact
                        to="/search"
                    >
                        Search
                    </NavLink>
                </div>

                <div className="ms-auto">
                    <ul className="navbar-nav">
                        <span className="nav-item nav-link text-info">
                            {user.name}
                        </span>
                        <button
                            className="btn nav-item nav-link"
                            onClick={handleLogout}
                        >
                            Logout
                        </button>
                    </ul>
                </div>
            </div>
        </nav>
    );
};
