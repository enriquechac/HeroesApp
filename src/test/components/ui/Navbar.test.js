import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router';
import { AuthContext } from '../../../auth/AuthContext';
import { Navbar } from '../../../components/ui/Navbar';
import { types } from '../../../types/types';

describe('Pruebas en el componente <Navbar/>', () => {
    const historyMock = {
        push: jest.fn(),
        location: {},
        replace: jest.fn(),
        listen: jest.fn(),
        createHref: jest.fn(),
    };

    const authValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Enrique',
        },
    };

    const wrapper = mount(
        <MemoryRouter>
            <AuthContext.Provider value={authValue}>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </AuthContext.Provider>
        </MemoryRouter>
    );

    afterEach(() => {
        jest.clearAllMocks();
    });

    test('Se debe de montar el componente correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Enrique');
    });

    test('Debe de llamar el logout y usar el history', () => {
        wrapper.find('button').prop('onClick')();
        expect(authValue.dispatch).toHaveBeenCalledTimes(1);
        expect(authValue.dispatch).toHaveBeenCalledWith({ type: types.logout });
        expect(historyMock.replace).toHaveBeenCalledWith('/login');
    });
});
