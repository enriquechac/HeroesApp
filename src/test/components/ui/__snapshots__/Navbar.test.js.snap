// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Navbar/> Se debe de montar el componente correctamente  1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
    <div
      className="ms-auto"
    >
      <ul
        className="navbar-nav"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Enrique
        </span>
        <button
          className="btn nav-item nav-link"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </div>
</nav>
`;
